dev/react-plugin-tools/bootstrap.py

```diff
@@ -45,55 +40,77 @@ def get_template_dir() -> Path:
         content = re.sub(license_pattern, "", content, flags=re.DOTALL)
-
     return content
 
-
-def copy_template_files(template_dir: Path, project_path: Path, project_name: str) -> None:
-    for item in template_dir.rglob("*"):
-        if item.is_file():
-            # Calculate relative path from template root
-            rel_path = item.relative_to(template_dir)
-            target_path = project_path / rel_path
-
-            target_path.parent.mkdir(parents=True, exist_ok=True)
-
-            with open(item, encoding="utf-8") as f:
-                content = f.read()
-
-            content = replace_template_variables(content, project_name)
-
-            file_extension = item.suffix.lower()
-            content = remove_apache_license_header(content, file_extension)
-
-            with open(target_path, "w", encoding="utf-8") as f:
-                f.write(content)
-
-            print(f"  Created: {rel_path}")
-
+def copy_and_process_file(src: Path, dest: Path, project_name: str) -> None:
+    dest.parent.mkdir(parents=True, exist_ok=True)
+    
+    with open(src, encoding="utf-8") as f:
+        content = f.read()
+    
+    content = replace_template_variables(content, project_name)
+    content = remove_apache_license_header(content, src.suffix.lower())
+    
+    with open(dest, "w", encoding="utf-8") as f:
+        f.write(content)
+
+def copy_directory_recursive(src_dir: Path, dest_dir: Path, project_name: str, skip_dirs: set = None) -> None:
+    """Recursively copy directory while processing files and skipping specified directories."""
+    if skip_dirs is None:
+        skip_dirs = set()
+    
+    for item in src_dir.iterdir():
+        # Skip directories that should be excluded
+        if item.is_dir() and item.name in skip_dirs:
+            print(f"✗ Skipping directory: {item.name}")
+            continue
+            
+        dest_item = dest_dir / item.name
+        
+        if item.is_dir():
+            dest_item.mkdir(parents=True, exist_ok=True)
+            copy_directory_recursive(item, dest_item, project_name, skip_dirs)
+        else:
+            # Process text files, copy binary files as-is
+            try:
+                copy_and_process_file(item, dest_item, project_name)
+            except UnicodeDecodeError:
+                # Binary file, copy as-is
+                shutil.copy2(item, dest_item)
+
+def copy_template_files(template_dir: Path, project_path: Path, project_name: str, include_ai_rules: bool) -> None:
+    
+    # Determine which directories to skip
+    skip_dirs = set()
+    if not include_ai_rules:
+        skip_dirs.add("ai_agents_rules")
+    
+    # Copy all files and directories, skipping excluded ones
+    copy_directory_recursive(template_dir, project_path, project_name, skip_dirs)
+    
+    # Provide feedback
+    if include_ai_rules:
+        ai_src = template_dir / "ai_agents_rules"
+        if ai_src.exists():
```

Comment:
Folder should exist, don't even handle the case. If it does not, we should raise an exception and stop.